apps/electric_migrations/lib/electric_migrations/ast/full_table_info.ex

```diff
@@ -0,0 +1,46 @@
+defmodule ElectricMigrations.Ast.FullTableInfo do
+  @moduledoc """
+  A struct to hold SQLite table info structure
+  """
+
+  alias ElectricMigrations.Ast
+
+  @enforce_keys [
+    :table_name,
+    :table_info,
+    :namespace
```

Comment:
They shouldn't be enforced, because this struct is built up iteratively instead of all at once. The "final" version of the struct will have the correct type of fields that won't have `nil`.